pkg/image/registry/image/etcd/etcd.go

```diff
@@ -59,8 +59,27 @@ func NewREST(optsGetter restoptions.Getter) (*REST, error) {
 	}
 
 	if err := restoptions.ApplyOptions(optsGetter, store, prefix); err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 
-	return &REST{store}, nil
+	internalStore := *store
+	internalStore.UpdateStrategy = image.InternalStrategy
+	internalStore.CreateStrategy = nil
+
+	return &REST{store}, &InternalREST{store: &internalStore}, nil
+}
+
+// InternalREST implements a RESTStorage for image resources internally modifiable.
```

Comment:
I disagree.  If someone can directly mutate the container (images), why
shouldn't they be allowed to control the signatures?

On Thu, Jul 14, 2016 at 1:55 PM, Michail Kargakis <notifications@github.com>
wrote:

> In pkg/image/registry/image/etcd/etcd.go
> <https://github.com/openshift/origin/pull/9181#discussion_r70852033>:
>
> >  	}
> >
> > -	return &REST{store}, nil
> > +	internalStore := *store
> > +	internalStore.UpdateStrategy = image.InternalStrategy
> > +	internalStore.CreateStrategy = nil
> > +
> > +	return &REST{store}, &InternalREST{store: &internalStore}, nil
> > +}
> > +
> > +// InternalREST implements a RESTStorage for image resources internally modifiable.
>
> Why can't I do all that using the current REST being returned? Seems like
> it has an update function, right?
>
> We should ignore updates on those fields in the normal path.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin/pull/9181/files/b1c344fd4d044ec643a5259b37924dcaff2e7ec3#r70852033>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AH2BSjZRauT9y30q26yYBXtGWBdj0MT4ks5qVngegaJpZM4IvEFx>
> .
>